python/mpt/src/mpt/utils.cairo

```diff
@@ -123,3 +124,50 @@ func deserialize_to_internal_node{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}
     );
     return result;
 }
+
+// @notice Checks if a branch node is valid.
+// A branch node is valid if it has at least two non-null subnodes.
+// @dev Raises an error if the branch node is invalid.
+// @param node The branch node to check.
+func check_branch_node(node: BranchNode) {
+    alloc_locals;
+
+    local first_non_null_branch;
+    local seconds_non_null_branch;
+    let subnodes_ptr = cast(node.value.subnodes.value, felt*);
+    %{
+        node = serialize(ids.node);
+        non_null_branches = [idx for idx, subnode in enumerate(node.subnodes) if (subnode is not None and subnode != b"")]
+        ids.first_non_null_branch, ids.seconds_non_null_branch = non_null_branches[0:2]
+    %}
+    // Check that the first subnode is not None and not empty
+    tempvar x = Extended(cast(subnodes_ptr[first_non_null_branch], ExtendedEnum*));
+
+    if (cast(x.value, felt) == 0) {
+        raise('ValueError');
+    }
+
+    let (buffer) = alloc();
+    tempvar empty_bytes = Bytes(new BytesStruct(data=buffer, len=0));
+
+    let invalid = Bytes__eq__(x.value.bytes, empty_bytes);
+
+    if (invalid.value != 0) {
+        raise('ValueError');
+    }
```

Comment:
use `assert_non_zero` on the pointer value. once asserted it's not zero, use `assert_non_zero` on the length of the bytes (as per the method described in the issue).

Reply:
assert_not_zero from sw does not work on pointers due to their call to `assert_integer`

```
// Verifies that value != 0. The proof will fail otherwise.
func assert_not_zero(value) {
    %{
        from starkware.cairo.common.math_utils import assert_integer
        assert_integer(ids.value)
        assert ids.value % PRIME != 0, f'assert_not_zero failed: {ids.value} = 0.'
    %}
    if (value == 0) {
        // If value == 0, add an unsatisfiable requirement.
        value = 1;
    }

    return ();
}
```